src/components/rangeslider/draw.js

```diff
@@ -279,17 +293,102 @@ function drawBg(rangeSlider, gd, axisOpts, opts) {
     });
 }
 
-function drawRangePlot(rangeSlider, gd, axisOpts, opts) {
-    var rangePlots = rangePlot(gd, opts._width, opts._height);
+function addClipPath(rangeSlider, gd, axisOpts, opts) {
+    var fullLayout = gd._fullLayout;
 
-    var gRangePlot = rangeSlider.selectAll('g.' + constants.rangePlotClassName)
+    var clipPath = fullLayout._topdefs.selectAll('#' + opts._clipId)
         .data([0]);
```

Comment:
Did you manage to factor out the `.data([0])` pattern? I suppose it would be a function with a name like `var clipPath = createOrAppend(fullLayout._topdefs, '#' + opts._clipId);`?